env.d.ts

```diff
@@ -764,46 +767,102 @@ type MongoQuerySelector<T extends MongoValue = MongoValue> = Partial<T extends M
 	(MongoArraySelectors<T> & MongoElementSelectors & MongoComparisonSelectors<T>) :
 	(MongoElementSelectors & MongoComparisonSelectors<T>)>
 
-type Query = { [key: string]: MongoValue | Query } & { _id?: Id, $in?: MongoValue[] }
-type Projection = Record<string, boolean | 0 | 1>
+type Query<Schema extends MongoSchema> = Partial<{[key in keyof Schema]: MongoValue | MongoQuerySelector<Schema[key]>}> & {
```

Comment:
I think you're looking for `{ [K in keyof Foo]?: Bar }` not `Partial<{ [K in keyof Foo]: Bar }>`.